File: benchmarks/java/integer_boolean/Java_Big_Integer/GreaterEqual.java
import de.hhu.stups.btypes.BInteger;
import de.hhu.stups.btypes.BBoolean;

public class GreaterEqual {






    private BInteger counter;

    public GreaterEqual() {
        counter = (BInteger) new BInteger("0");
    }

    public void simulate() {
        while((counter.less(new BInteger("5000000")).and(new BInteger("2").greaterEqual(new BInteger("1")))).booleanValue()) {
            counter = (BInteger) counter.plus(new BInteger("1"));
        }
    }

    public static void main(String[] args) {
        GreaterEqual exec = new GreaterEqual();
        long start = System.nanoTime();
        exec.simulate();
        long end = System.nanoTime();
        System.out.println(exec.getClass().toString() + " Execution: " + (end - start));
    }

}

File: benchmarks/cpp/integer_boolean/primitive_integer/GreaterEqual.cpp
#include <iostream>
#include <string>
#include "BUtils.cpp"
#include "BInteger.cpp"
#include "BBoolean.cpp"

#ifndef GreaterEqual_H
#define GreaterEqual_H

using namespace std;

class GreaterEqual {



    private:



        BInteger counter;

    public:

        GreaterEqual() {
            counter = (BInteger(0));
        }

        void simulate() {
            while((counter.less((BInteger(5000000)))._and((BInteger(2)).greaterEqual((BInteger(1))))).booleanValue()) {
                counter = counter.plus((BInteger(1)));
            }
        }

};
int main() {
    clock_t start,finish;
    double time;
    GreaterEqual exec;
    start = clock();
    exec.simulate();
    finish = clock();
    time = (double(finish)-double(start))/CLOCKS_PER_SEC;
    printf("%f\n", time);
    return 0;
}
#endif

